### PR TITLE
Feature/df 1597 avam occupations importer part 1

### PR DESCRIPTION
**Pull request by emina-str**
DONE:
- merged with current develop branch (lots of merge conflicts resolved)
- move enums (Language, ProfessionCodeType, etc.) to enum package (changes in import statements)
- change return value of `OccupationLabelMappingRepository.findOneByBfsCode` (before it was returning a List, now an `Optional<OccupationLabelMapping>`), doesn't seem right to return a List, as Bfs should be unique
- AvamOccupationBatchImporter (implementation and tests)
- added AvamOccupation table (plus liquibase script)
- added event while saving AvamOccupation (but will be finished in second PR with DomainEventSenderGateway with handling OccupationLabel and OccupationLabelMapping)
- build.gradle dependencies

### CODE REVIEW

File: avam-occupations-importer-app/src/main/resources/bootstrap.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# ===================================================================
+# Spring Cloud Config bootstrap configuration for the "dev" profile
+# In prod profile, properties will be overwriten by the ones defined in bootstrap-prod.yml
+# ===================================================================
+
+spring:
+    application:
+        name: avam-occupations-importer
+#    cloud:
```
Comment (fabiangotzen) [inline avam-occupations-importer-app/src/main/resources/bootstrap.yml]: I think this is needed for spring cloud task to work.

File: avam-occupations-importer-app/src/test/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/BatchOccupationsReaderTest.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
+
+@SpringBootTest
+@ExtendWith(SpringExtension.class)
```
Comment (fabiangotzen) [inline avam-occupations-importer-app/src/test/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/BatchOccupationsReaderTest.java]: I'm not sure we need the @ExtendWith annotation

File: avam-occupations-importer-app/src/test/resources/application-test.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    batch:
+        initialize-schema: always
+        job:
+            enabled: false # Spring automatically runs batch jobs, to prevent two batch jobs from running when a test is started, we need to disable this.
+    datasource:
+        url: jdbc:h2:mem:batch;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=false
+        username: batch
+        password:
+        platform: h2
+    cloud:
```
Comment (fabiangotzen) [inline avam-occupations-importer-app/src/test/resources/application-test.yml]: I think this config property is not needed for the tests to work. (cloud.stream.bindings.output.destination)

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd">
+    <changeSet author="liquibase-docs" id="loadData-example">
+        <loadData file="../data/avam_beruf.csv" catalogName="intermediate" schemaName="public" tableName="avam_beruf" separator="," relativeToChangelogFile="true">
+            <column name="cfa_beruf_avam" type="NUMERIC"/>
```
Comment (fabiangotzen) [inline avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: These column definitions are redundant (they are already done in the initialize_schema). All we need to do is to load the data. 
IMO its unnecessary to use liquibase in these tests. We can just let spring initialize the schema for us and then import the data as data.sql (You can create this .sql by importing the .csv in the docker table and then create the "Insert Into statements" in Intellij's sql editor).  
It's not a mandatory change but I think it would be a more elegant and easier to change solution.

File: avam-occupations-importer-app/src/test/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/BatchOccupationsReaderTest.java (hunk 2 above)
Comment (emina-str) [reply avam-occupations-importer-app/src/test/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/BatchOccupationsReaderTest.java]: not needed

File: avam-occupations-importer-app/src/test/resources/application-test.yml (hunk 3 above)
Comment (emina-str) [reply avam-occupations-importer-app/src/test/resources/application-test.yml]: not needed

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/AvamOccupationDto.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
+package ch.admin.seco.service.reference.service.dto;
+
+import ch.admin.seco.service.reference.domain.AvamOccupation;
+
+import java.time.LocalDate;
+
+public class AvamOccupationDto {
+
+    private String avamCode;
```
Comment (fabiangotzen) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/AvamOccupationDto.java]: I think we could add the same @NotBlank validation for avamcode/ bfscode /cfacode here?

File: reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    <changeSet id="20191008180000-2" author="haso">
+        <createTable tableName="avam_occupation">
+            <column name="cfa_code" type="varchar(20)">
+                <constraints nullable="false" unique="true"/>
+            </column>
+            <column name="bfs_code" type="varchar(20)"/>
+            <column name="avam_code" type="varchar(20)"/>
+            <column name="valid_from" type="DATE"/>
+            <column name="valid_to" type="DATE"/>
+            <column name="feminine_text_de" type="varchar(255)"/>
```
Comment (fabiangotzen) [inline reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml]: I think we used varchar(1000) in the other liqubiasescripts. I think we should use 255 everywhere. (Since its just a title of a beruf 255 should be enough)

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java [hunk 7]
```diff
@@ -90,16 +89,14 @@ private ReportingObligationDTO checkReportingObligation(String cantonCode, Profe
             .professionCode(professionCodeDTO);
         if (featureToggleProperties.isChIscoEnabled()) {
             log.debug("Request to check reportingObligation profession code by ChIscoCode : {}", professionCodeDTO.getCode());
-            reportingObligationRepository.findOneByChIsco5Code(professionCodeDTO.getCode()).ifPresent(obligation -> {
-                reportingObligationDTO.hasReportingObligation(obligation.isAppliedForSwissOrCanton(cantonCode))
-                    .cantons(obligation.getCantonCodes());
-            });
+            reportingObligationRepository.findOneByChIsco5Code(professionCodeDTO.getCode())
```
Comment (seco-mgabor) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java]: I had to re-factor this ugly code, sorry.

File: reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+    xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+    <changeSet id="20191008180000-2" author="haso">
+        <createTable tableName="avam_occupation">
+            <column name="cfa_code" type="varchar(20)">
+                <constraints nullable="false" unique="true"/>
+            </column>
+            <column name="bfs_code" type="varchar(20)"/>
+            <column name="avam_code" type="varchar(20)"/>
+            <column name="valid_from" type="DATE"/>
```
Comment (seco-mgabor) [inline reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml]: valid_from must not be nullable.

File: reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventReceiverGatewayTest.java [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    private AvamOccupationRepository avamOccupationRepository;
+
+    @Autowired
+    private AvamOccupationEventReceiverGateway avamOccupationEventReceiverGateway;
+
+    @MockBean
+    private DomainEventPublisher domainEventPublisher;
+
+    @Before
+    public void setUp() {
```
Comment (seco-mgabor) [inline reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventReceiverGatewayTest.java]: Why do you use "this" for fields, which have clear names and scopes? It's just boiler plate code.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/OccupationLabelServiceImpl.java [hunk 10]
```diff
@@ -230,7 +219,7 @@ private OccupationLabel getBestMatchingOccupationLabel(final ProfessionCodeDTO p
             case AVAM:
                 return occupationMappingRepository.findOneByAvamCode(code);
             case BFS:
-                return occupationMappingRepository.findByBfsCode(code).stream().findFirst();
+                return occupationMappingRepository.findOneByBfsCode(code);
```
Comment (seco-mgabor) [inline reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/OccupationLabelServiceImpl.java]: I'm still not sure about this change. Right now, the data structure allows more than 1 record as result for this query.
When it's suppose to return just one record, we should enforce it also in the DB structure.

File: reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerConfig.java [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+    @EnableBinding(MessageBrokerChannels.class)
+    static class DefaultMessageBroker {
+    }
+
+    @Configuration
+    @Profile(MESSAGE_BROKER_MOCK)
+    @EnableBinding(MessageBrokerChannels.class)
+    static class MockedMessageBroker {
+
+        @Bean
```
Comment (seco-mgabor) [inline reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerConfig.java]: Replace this by method reference:
`@Bean
        MessageBrokerChannels messageBrokerChannels() {
            return DirectChannel::new;
        }
    }`

File: reference-app/src/main/java/ch/admin/seco/service/reference/domain/occupation/OccupationLabelMappingRepository.java [hunk 12]
```diff
@@ -21,7 +19,7 @@
     @Query("select o from OccupationLabelMapping o where o.avamCode in (select m.avamCode from OccupationLabelMappingX28 m where m.x28Code = ?1)")
     Optional<OccupationLabelMapping> findOneByX28Code(String x28Code);
 
-    List<OccupationLabelMapping> findByBfsCode(String bfsCode);
+    Optional<OccupationLabelMapping> findOneByBfsCode(String bfsCode);
```
Comment (seco-mgabor) [inline reference-app/src/main/java/ch/admin/seco/service/reference/domain/occupation/OccupationLabelMappingRepository.java]: Why did you change this?

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml (hunk 4 above)
Comment (seco-mgabor) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: I agree with the comment about redundancy.
But the docker stuff is not in the test's scope. The tests must be self contained and not depending on other environments, outside their scope.
I don't see anything wrong with liquibase and this is used widely for integration tests in other projects as well.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/OccupationEventProcessor.java [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+
+    @Override
+    public AvamOccupationEventDto process(AvamOccupationDto avamOccupationDto) {
+        AvamOccupationEventDto avamOccupationEventDto = new AvamOccupationEventDto(
+            avamOccupationDto, AvamOccupationEventDtoType.AVAM_OCCUPATION_IMPORTED);
+        Set<ConstraintViolation<AvamOccupationEventDto>> violations = validator.validate(avamOccupationEventDto);
+        if (violations.isEmpty()) {
+            return avamOccupationEventDto;
+        }
+        return null;
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/OccupationEventProcessor.java]: No, it is not. If it is null, it is only not send to writer.

[the remark this answers is not in the document]
Comment (seco-mgabor) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/OccupationEventProcessor.java]: Sorry, I didn't read the ItemProcessor JavaDoc. You're right.

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml (hunk 4 above)
Comment (fabiangotzen) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: This has nothing to do with docker. I'm saying that we can just create the schema & data loading for the h2 db with a simple data.sql in the ressources folder. There is nothing "wrong" with using liquibase for this but it's in my opinion just not necessary.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/MessageHeaders.java [hunk 14]
```diff
@@ -0,0 +1,12 @@
+package ch.admin.seco.service.reference.infrastructure.batch.occupation.importer.config;
+
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/MessageHeaders.java]: `MessageHeaders` class is our code style for basically all projects. It will not be changed.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/OccupationMessageChannelWriter.java [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Override
+    public void write(List<? extends AvamOccupationEventDto> items) {
+        items.forEach(this::writeItem);
+    }
+
+    private void writeItem(@Valid AvamOccupationEventDto avamOccupationEventDto) {
+        this.messageChannel.send(
+                MessageBuilder.withPayload(avamOccupationEventDto)
+                    .setHeader(MessageHeaders.PARTITION_KEY, avamOccupationEventDto.getAvamOccupationDto().getCfaCode())
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/OccupationMessageChannelWriter.java]: Writing it like so is our code style for basically all projects. It will not be changed.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/model/AvamOccupationDto.java [hunk 16]
```diff
@@ -0,0 +1,114 @@
...
+
+    @NotBlank
+    private String cfaCode;
+
+    @NotBlank
+    private String avamCode;
+
+    @NotBlank
+    private String bfsCode;
+
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/model/AvamOccupationDto.java]: Constraints should be (and are) on the fields we actually use and need for further processing, this is not the case for validFrom.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java (hunk 7 above)
Comment (fabiangotzen) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java]: A couple of things: 
1) We don't push to  each others Pullrequests without permission to do so. It is not nice and not how we work as a team. Please also use constructive language in Reviews. Most of the time there is a reason for "ugly code". Which brings me to 2)

2) It was a conscious and pragmatic decision to simply duplicate this piece because this is Throwaway Code and was planned to be removed in this task: https://alv-ch.atlassian.net/browse/DF-1915 

3) I don't mind the refactoring itself. Boyscouting is a good habit. But you have got to understand that this is quite a crucial class regarding JobRoom and one of the deciding things for a successful Migration to CH-ISCO. If this change of yours would introduce a bug by refactoring this in a completly unrelated PR without me noticing this would be incredibly hard for me to troubleshoot during the migration.

So please remove this change from this PR. It is not needed since I'll remove it anyway in one month.

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/initialize_schema.xml [hunk 17]
```diff
@@ -0,0 +1,21 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.6.xsd">
+
+    <changeSet author="fago (generated)" id="1560777687162-5">
+        <createTable tableName="avam_beruf">
+            <column name="cfa_beruf_avam" type="numeric"/>
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/initialize_schema.xml]: Constraints should be (and are) on the fields we actually use and need for further processing, this is not the case for validFrom.

File: reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml (hunk 8 above)
Comment (emina-str) [reply reference-app/src/main/resources/config/liquibase/changelog/20191108180000_add_avam_occupation.xml]: Constraints should be (and are) on the fields we actually use and need for further processing, this is not the case for validFrom.

File: reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerConfig.java (hunk 11 above)
Comment (emina-str) [reply reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerConfig.java]: existing code style.

File: reference-app/src/main/java/ch/admin/seco/service/reference/domain/occupation/OccupationLabelMappingRepository.java (hunk 12 above)
Comment (emina-str) [reply reference-app/src/main/java/ch/admin/seco/service/reference/domain/occupation/OccupationLabelMappingRepository.java]: reverted.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/OccupationLabelServiceImpl.java (hunk 10 above)
Comment (emina-str) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/occupation/OccupationLabelServiceImpl.java]: reverted.

File: reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java (hunk 7 above)
Comment (seco-mgabor) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java]: 1) I used this method in other companies and it worked just fine.
2) I think that every developer should write the best code (s)he can and I expecting that too.
Dirty hacks and ugly code can exist locally, but not on master or develop branches.
3) If that feature is so important, you should have written a test for it. If am not mistaking, the existing tests for this class are from 2018.

Comment (seco-mgabor) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java]: I've reverted the changes, but the integration tests have to be extended to cover CH ISCO use case.

Comment (fabiangotzen) [reply reference-app/src/main/java/ch/admin/seco/service/reference/service/reportingobligation/ReportingObligationServiceImpl.java]: 1. as discussed
2. as discussed
3. At the time this was merged we did not have the needed data from the business to determine the reporting obligation. But you are right on this, this is still a todo for a seperate branch.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java [hunk 18]
```diff
@@ -0,0 +1,95 @@
...
+
+import javax.sql.DataSource;
+import javax.validation.Validator;
+
+@Configuration
+@EnableConfigurationProperties(AvamOccupationImporterProperties.class)
+public class AvamOccupationImporterBatchConfiguration {
+
+    private static final String SELECT_AVAM_OCCUPATIONS =
+        "SELECT cfa_beruf_avam                  as cfaCode," +
```
Comment (birom) [inline avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: What does mean ``cfaCode`` can not we call it everywhere ``externalId``?

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml (hunk 4 above)
Comment (birom) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: In other projects we use the existing Liquibase scripts, now we are developing one only for the tests. 
I do not see any benefit of using Liquibase in a temporary test DB either.
If you want to have a "self-contained" tests have a look at the [spring-testing-annotation-sql](https://docs.spring.io/spring-framework/docs/current/spring-framework-reference/testing.html#spring-testing-annotation-sql).

File: reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventDto.java [hunk 19]
```diff
@@ -0,0 +1,28 @@
+package ch.admin.seco.service.reference.infrastructure.messagebroker.avamoccupation;
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventDto.java]: The package name should be ``ch.admin.seco.service.reference.infrastructure.messagebroker.occupation`` because the ``AvamOccupationService`` is also in an ``occupation`` sub package.

File: reference-app/src/main/java/ch/admin/seco/service/reference/domain/AvamOccupation.java [hunk 20]
```diff
@@ -0,0 +1,210 @@
+package ch.admin.seco.service.reference.domain;
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/domain/AvamOccupation.java]: The class should be in a ``ch.admin.seco.service.reference.domain.occupation`` package.

File: reference-app/src/main/java/ch/admin/seco/service/reference/domain/AvamOccupationRepository.java [hunk 21]
```diff
@@ -0,0 +1,11 @@
+package ch.admin.seco.service.reference.domain;
```
Comment (birom) [inline reference-app/src/main/java/ch/admin/seco/service/reference/domain/AvamOccupationRepository.java]: The class should be in a ``ch.admin.seco.service.reference.domain.occupation`` package.

File: reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventDto.java (hunk 19 above)
Comment (emina-str) [reply reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/avamoccupation/AvamOccupationEventDto.java]: done

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml (hunk 4 above)
Comment (emina-str) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: will change as suggested.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java [hunk 22]
```diff
@@ -0,0 +1,95 @@
...
+            "   dat_gueltig_ab                  as validFrom," +
+            "   dat_gueltig_bis                 as validTo," +
+            "   txl_beruf_w_de                  as feminineTextDe," +
```
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: will change.

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java (hunk 18 above)
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: will change.

Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: changed to `externalId`

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java (hunk 22 above)
[the remark this answers is not in the document]
Comment (emina-str) [reply avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: changed.

File: reference-app/src/main/java/ch/admin/seco/service/reference/domain/events/DomainEventType.java [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+
+    DomainEventType(String value) {
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    @Override
```
[the remark this answers is not in the document]
Comment (dvoracek-martin) [reply reference-app/src/main/java/ch/admin/seco/service/reference/domain/events/DomainEventType.java]: what

File: reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerChannels.java [hunk 24]
```diff
@@ -0,0 +1,13 @@
+package ch.admin.seco.service.reference.infrastructure.messagebroker;
+
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.messaging.SubscribableChannel;
+
+public interface MessageBrokerChannels {
+
```
[the remark this answers is not in the document]
Comment (dvoracek-martin) [reply reference-app/src/main/java/ch/admin/seco/service/reference/infrastructure/messagebroker/MessageBrokerChannels.java]: make it final

File: avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml (hunk 4 above)
Comment (emina-str) [reply avam-occupations-importer-app/src/test/resources/liquibase/changelog/load_data.xml]: done. please check.

File: reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/AvamOccupationFixture.java [hunk 25]
```diff
@@ -0,0 +1,57 @@
+package ch.admin.seco.service.reference.infrastructure.messagebroker;
+
+import ch.admin.seco.service.reference.domain.occupation.AvamOccupation;
+
+import java.time.LocalDate;
+
+public class AvamOccupationFixture {
+
+    private static final String CFC_CODE = "cfcCode";
```
Comment (fabiangotzen) [inline reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/AvamOccupationFixture.java]: I think this should also be renamed to externalId

File: avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java [hunk 26]
```diff
@@ -0,0 +1,95 @@
...
+                                                    AvamOccupationImporterProperties avamOccupationImporterProperties,
+                                                    MessageChannel output,
+                                                    Validator validator,
+                                                    @Qualifier("avamDataSource") DataSource avamDataSource) {
+        this.avamOccupationImporterProperties = avamOccupationImporterProperties;
+        this.avamDataSource = avamDataSource;
+        this.jobBuilderFactory = jobBuilderFactory;
+        this.stepBuilderFactory = stepBuilderFactory;
+        this.output = output;
+        this.validator = validator;
```
Comment (fabiangotzen) [inline avam-occupations-importer-app/src/main/java/ch/admin/seco/service/reference/infrastructure/batch/occupation/importer/config/AvamOccupationImporterBatchConfiguration.java]: I think we should also add a maillistener for this Batchjob, so that an email will be sent in case the Job should fail. We have existing solutions for this in other batchjobs. But that can also be added in part2.

File: reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/AvamOccupationFixture.java (hunk 25 above)
Comment (emina-str) [reply reference-app/src/test/java/ch/admin/seco/service/reference/infrastructure/messagebroker/AvamOccupationFixture.java]: fixed
